### ISSUE TITLE
Editor: Move any editor code in client/post-editor that is needed elsewhere

### ISSUE THREAD

**Issue by allilevine** [Automattic/wp-calypso#46367]
## Requirements

Now that the [WordPress.com editor is deprecated](https://github.com/Automattic/wp-calypso/issues/41086), we're removing all unneeded code related to the Calypso editor and the editor preference. Some of the code in client/post-editor is still needed elsewhere in the app, so we'd like to move it out of that folder.

### To move:

*Used by Gutenberg (client/gutenberg/editor/calypsoify-iframe.tsx)*
- [ ] **EditorDocumentHead**
- [ ] **EditorRevisionsDialog** (which needs **EditorRevisionsList** and **EditorDiffViewer**)

*Used by My Sites > Media (client/my-sites/media/main.jsx)*
- [ ] **EditorMediaModalDialog** in main.jsx
- [ ] **EditorMediaModalDetail** in main.jsx
- [ ] **DetailPreviewVideo** in client/blocks/video-editor/index.jsx  (only used in my-sites/media)

*Used by My Sites > Site Settings*
- [ ] **EditorMediaModalDialog** is used in 
    - client/my-sites/site-settings/podcast-cover-image-setting/index.jsx
    - client/my-sites/site-settings/site-icon-setting/index.jsx
- [ ] **TermSelectorAddTerm** is used in client/blocks/term-tree-selector/add-term.jsx, client/blocks/term-tree-selector/index.jsx (could be replaced with `TermFormDialog`) -- used by Site Settings > Writing > Podcasting

*Used by My Sites > Posts*
- [x] **PublicizeMessage** is used in client/blocks/post-share/index.jsx (post sharing from /posts) and needs **EditorDrawerLabel** and **EditorFieldset** (PR: #47095)

*Parent Component Not In Use*
- **EditorFeaturedImagePreviewContainer** is used in client/components/tinymce/plugins/simple-payments/dialog/product-image-picker.jsx and needs **EditorMediaModal / MediaModal**
- **EditorHtmlToolbar** is used in client/components/tinymce/index.jsx and needs **Markup / MediaMarkup** and **EditorMediaModal / MediaModal**
- **EditorMediaModal / MediaModal** is used in 
    - client/components/tinymce/plugins/media/plugin.jsx
- **Markup / MediaMarkup** is used in
    - client/components/tinymce/plugins/media/drop-zone.jsx
    - client/components/tinymce/plugins/media/plugin.jsx
- **EditorMediaModal / MediaModal** is used in 
    - client/blocks/image-selector/index.jsx (may not be used anywhere)
- **EditorMediaModalGalleryHelp** is used in client/my-sites/media-library/list-item.tsx (used in tests and post-editor)
- **MediaModalSecondaryActions** is used in client/my-sites/media-library/header.jsx (used in tests)

**Comment by worldomonation** [comment]
Hi! I'm performing triage on older bugs as part of Quality Squad's efforts to cut down on backlog issue counts.

As there has been no recent activity on this issue it is being closed with the label `Closed - Won't Fix`.
If this issue should remain open, please feel free to re-open this bug.

**Comment by allilevine** [comment]
Thanks @worldomonation. I think this issue may still be needed because there's still editor code in client/post-editor. 

@tyxla Is that correct?

**Comment by tyxla** [comment]
Yep. I think we should still move all that lives in `client/post-editor` to respective directories and delete that one. Maybe an easy way is to just move the entire `post-editor` to `client/components/post-editor`?

**Comment by allilevine** [comment]
> Yep. I think we should still move all that lives in `client/post-editor` to respective directories and delete that one. 

Thanks, reopening!

> Maybe an easy way is to just move the entire `post-editor` to `client/components/post-editor`?

That does sound easier. Will that create technical debt though? Will we have to move them out of there eventually?

**Comment by tyxla** [comment]
> > Maybe an easy way is to just move the entire `post-editor` to `client/components/post-editor`?
> 
> That does sound easier. Will that create technical debt though? Will we have to move them out of there eventually?

I don't think it would create any technical debt, because all that remains there is actually components indeed. The only thing to reason about is whether each of them should be in `client/components` or in `client/blocks`, as some of them are pretty convoluted (the media modal for example).
